Formula/glibc.rb

```diff
@@ -95,41 +95,162 @@ class Glibc < Formula
 
   depends_on "binutils" => :build
   depends_on GawkRequirement => :build
-  depends_on "linux-headers@4.4" => :build
+  depends_on "linux-headers@5.15" => :build
   depends_on MakeRequirement => :build
   depends_on SedRequirement => :build
   depends_on BrewedGlibcNotOlderRequirement
   depends_on :linux
   depends_on LinuxKernelRequirement
 
-  # GCC 4.7 or later is required.
-  fails_with gcc: "4.6"
+  # Vendor build dependencies that are needed because the host versions
+  # may be unavailable or too old.  Resources should be only updated
+  # when their version in the INSTALL file exceeds the version here.
+  # Example INSTALL file: https://elixir.bootlin.com/glibc/glibc-2.35/source/INSTALL.
+  resource "make" do
+    url "https://ftp.gnu.org/gnu/make/make-4.3.tar.gz"
+    sha256 "e05fdde47c5f7ca45cb697e973894ff4f5d79e13b750ed57d7b66d8defc78e19"
+  end
 
-  def install
-    # Fix Error: `loc1@GLIBC_2.2.5' can't be versioned to common symbol 'loc1'
-    # See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=869717
-    # Fixed in glibc 2.24
-    inreplace "misc/regexp.c", /^(char \*loc[12s]);$/, "\\1 __attribute__ ((nocommon));"
+  resource "python3" do
+    url "https://www.python.org/ftp/python/3.10.5/Python-3.10.5.tgz"
+    sha256 "18f57182a2de3b0be76dfc39fdcfd28156bb6dd23e5f08696f7492e9e3d0bf2d"
+  end
+
+  resource "m4" do
+    url "https://ftp.gnu.org/gnu/m4/m4-1.4.19.tar.xz"
+    sha256 "63aede5c6d33b6d9b13511cd0be2cac046f2e70fd0a07aa9573a04a82783af96"
+  end
+
+  resource "bison" do
+    url "https://ftp.gnu.org/gnu/bison/bison-3.8.2.tar.xz"
+    sha256 "9bba0214ccf7f1079c5d59210045227bcf619519840ebfa80cd3849cff5a5bf2"
+  end
+
+  resource "bzip2" do
+    url "https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz"
+    sha256 "ab5a03176ee106d3f0fa90e381da478ddae405918153cca248e682cd0c4a2269"
+  end
+
+  resource "gcc" do
+    url "https://ftp.gnu.org/gnu/gcc/gcc-10.4.0/gcc-10.4.0.tar.xz"
+    sha256 "c9297d5bcd7cb43f3dfc2fed5389e948c9312fd962ef6a4ce455cff963ebe4f1"
+  end
+
+  resource "mpfr" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/mpfr-3.1.6.tar.bz2"
+    sha256 "cf4f4b2d80abb79e820e78c8077b6725bbbb4e8f41896783c899087be0e94068"
+  end
 
+  resource "mpc" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/mpc-1.0.3.tar.gz"
+    sha256 "617decc6ea09889fb08ede330917a00b16809b8db88c29c31bfbb49cbf88ecc3"
+  end
+
+  resource "gmp" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/gmp-6.1.0.tar.bz2"
+    sha256 "498449a994efeba527885c10405993427995d3f86b8768d8cdf8d9dd7c6b73e8"
+  end
```

Comment:
Last I recall, we added these resources to avoid needing `wget`, because the GCC script that handles these dependencies uses `wget`.

Could we consider using a `WgetRequirement` instead? That seems fairly simple enough for users to install on their own, and I don't imagine the version requirements are particularly stringent.

That is, of course, as long as the GCC script that fetches these dependencies are versioned (which it almost certainly is) and checksummed (likely, but important to verify).

Reply:
Even if they aren't versioned or checksumed: it seems we're not going to end up bumping these ourselves anyway.

I'd also consider just doing `depends_on "wget"` as we're not going to link against it.

Reply:
Actually, it doesn't even look like `wget` is required at all:
```sh
# contrib/download_prerequisites
if type wget > /dev/null ; then
  fetch='wget'
else
  fetch='curl -LO'
fi
```

Reply:
Oops I need to check why this wasn’t working when `wget` isn’t installed.

Reply:
But: one advantage of vendoring these resources ourselves is that we can then choose tarballs that don't require `bzip2`. We're not taking advantage of that at the moment, but should be possible.

Reply:
I've pushed a commit to use the bundled script, which does work with `curl`.  I think I was using an older version of GCC originally that hardcoded `wget`, but the new versions don't.  The official upstream for GCC only releases `bz2` archives for the libraries that need `bzip2`, and I'd be wary of using unofficial sources just to get the archives in a different format.  `bzip2` is trivial to vendor so I think it's worth not having to vendor the GCC support libraries.